Compiler warning: potential null pointer dereference

### What is the issue?

```
src/transmission/libtransmission/torrent.cc: In member function ‘std::string_view tr_torrent::primaryMimeType() const’:
src/transmission/libtransmission/torrent.cc:2320:16: warning: potential null pointer dereference [-Wnull-dereference]
 2320 |     return it->first;
      |                ^~~~~
src/transmission/libtransmission/torrent.cc:2320:16: warning: potential null pointer dereference [-Wnull-dereference]

```

```
In member function ‘tr_tracker* tr_tier::currentTracker()’,
    inlined from ‘tr_tracker* tr_tier::useNextTracker()’ at src/transmission/libtransmission/announcer.cc:419:30,
    inlined from ‘void {anonymous}::on_scrape_done_helpers::on_scrape_error(const tr_session*, tr_tier*, const char*)’ at src/transmission/libtransmission/announcer.cc:1245:43:
src/transmission/libtransmission/announcer.cc:357:20: warning: potential null pointer dereference [-Wnull-dereference]
  357 |             return nullptr;
      |                    ^~~~~~~
src/transmission/libtransmission/announcer.cc: In function ‘void {anonymous}::multiscrape(tr_announcer_impl*, const std::vector<tr_tier*>&)’:
src/transmission/libtransmission/announcer.cc:1404:65: warning: potential null pointer dereference [-Wnull-dereference]
 1404 |         auto const* const scrape_info = tier->currentTracker()->scrape_info;
      |   
```


### Which application of Transmission?

None

### Which version of Transmission?

4.0.0

--- a/libtransmission/announcer.cc
+++ b/libtransmission/announcer.cc
@@ -1401,10 +1401,21 @@ void multiscrape(tr_announcer_impl* announcer, std::vector<tr_tier*> const& tier
     // batch as many info_hashes into a request as we can
     for (auto* tier : tiers)
     {
-        auto const* const scrape_info = tier->currentTracker()->scrape_info;
-        bool found = false;
+        auto const* const current_tracker = tier->currentTracker();
+        TR_ASSERT(current_tracker != nullptr);
+        if (current_tracker == nullptr)
+        {
+            continue;
+        }
 
+        auto const* const scrape_info = current_tracker->scrape_info;
         TR_ASSERT(scrape_info != nullptr);
+        if (scrape_info == nullptr)
+        {
+            continue;
+        }
+
+        bool found = false;
 
         /* if there's a request with this scrape URL and a free slot, use it */
         for (size_t j = 0; !found && j < request_count; ++j)

